ngraph/src/ngraph/op/proposal.cpp

```diff
@@ -121,3 +121,78 @@ bool AttributeAdapter<op::ProposalAttrs>::visit_attributes(AttributeVisitor& vis
     visitor.on_attribute("framework", m_ref.framework);
     return true;
 }
+
+constexpr NodeTypeInfo op::v4::Proposal::type_info;
+
+op::v4::Proposal::Proposal(const Output<Node>& class_probs,
+                           const Output<Node>& class_bbox_deltas,
+                           const Output<Node>& image_shape,
+                           const op::ProposalAttrs& attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v4::Proposal::validate_and_infer_types()
+{
+    set_input_is_relevant_to_shape(2);
+
+    const auto& class_probs_pshape = get_input_partial_shape(0);
```

Comment:
Can we call the `v0::Proposal::validate_and_infer_types` first to perform generic checks. And then infer the second output specific for `v4::Proposal`? This way we reduce the amount of code significantly.

Reply:
Good idea. Removed overriding of `validate_and_infer_types` in v4. Second output types/shapes are processed in constructor of v4. Please review.

Reply:
This is not correct. We should override the `validate_and_infer_types` in the v1 and call the base class method. Because the `validate_and_infer_types` may be called several times and it is not enough to run it it in the constructor.

Reply:
I see. I will change that.

Reply:
Changed. Please review.